src/Renci.SshNet/Channels/Channel.cs

```diff
@@ -554,9 +554,9 @@ protected virtual void Close()
                         // SSH_MSG_CHANNEL_CLOSE as response to a SSH_MSG_CHANNEL_CLOSE sent by the
                         // server
                         var closeWaitResult = _session.TryWait(_channelClosedWaitHandle, ConnectionInfo.ChannelCloseTimeout);
-                        if (closeWaitResult != WaitResult.Success)
+                        if (closeWaitResult != WaitResult.Success && Diagnostics.IsEnabled(TraceEventType.Warning))
```

Comment:
I don't get why we need the condition `Diagnostics.IsEnabled(TraceEventType.Warning)` outside the log method. 

what if someone forgot to check it?

Reply:
We don't need it, it just prevents doing any extra work like allocating string instances unnecessarily in the case that the log level is not enabled. But TBH I don't think it would make a material impact either way.